app/assets/stylesheets/application.css

```diff
@@ -0,0 +1,15 @@
+/*
+ * This is a manifest file that'll be compiled into application.css, which will include all the files
+ * listed below.
+ *
+ * Any CSS and SCSS file within this directory, lib/assets/stylesheets, or any plugin's
+ * vendor/assets/stylesheets directory can be referenced here using a relative path.
+ *
+ * You're free to add application-wide styles to this file and they'll appear at the bottom of the
+ * compiled file so the styles you add here take precedence over styles defined in any other CSS/SCSS
+ * files in this directory. Styles in this file should be added after the last require_* statement.
+ * It is generally better to create a new file per style scope.
+ *
+ *= require_tree .
+ *= require_self
+ */
```

Comment:
1. Here in the comments say that this will compile all the files listed into application.css. How is that done? Theoretically & practically

Reply:
The Rails app uses a gem called sprockets to concatenate, or compile, all of the CSS files into this master .css file. This allows you to then define separate stylesheets either in the same directory or in different parts of the app to keep things more organised.

During the compilation step, all these files are then combined together by appending one after another to the master file.

Reply:
> This allows you to then define separate stylesheets either in the same directory or in different parts of the app to keep things more organised.

Where is the code that define the above?

Reply:
Do you mean like how we can write this example in code?

If we have another .css file called `file-1.css`, we can add the line `*= require file-1` to import its CSS.

Reply:
Yes I'm looking for code. And your answer is wrong. You don't need to specifically add a `require` again. Read the comments. Its already saying it does include all. I am asking where does it define that to be able to include all?

Reply:
The `*= require_tree .` is responsible for compiling all the css files in this directory together. Any new `require` statements can be added if we need to also include any other css files that are declared in other directories, such as those in `vendor/assets/stylesheets` by referencing them using relative paths.